src/app.py

```diff
@@ -42,8 +49,8 @@
 print ("debug :" + str(result.json()))
 
 with open(iexec_out + '/result.txt', 'w+') as f:
-    if str(result.status_code) == "200":
-        f.write("email send successfully")
+    if not dataset_exists or not str(result.status_code) == "200":
```

Comment:
```
if code == 200
else write failed
```
looks enough (initialized to 0 on L12)

Reply:
Right, see https://github.com/iExecBlockchainComputing/confidential-email-sender/pull/3/commits/f12a596e3e77ba12978eab071beeaf6da5f59bfa.

Reply:
I don't see the added value of the `not`, but it is great otherwise, thanks :)